packages/e2e-tests/tests/chat-list-multiselect.spec.ts

```diff
@@ -0,0 +1,481 @@
+import { test, expect, type Page, type Locator } from '@playwright/test'
+
+import {
+  createProfiles,
+  User,
+  loadExistingProfiles,
+  deleteAllProfiles,
+  reloadPage,
+} from '../playwright-helper'
+
+test.describe.configure({ mode: 'serial' })
+
+expect.configure({ timeout: 5_000 })
+test.setTimeout(30_000)
+
+let existingProfiles: User[] = []
+
+const numberOfProfiles = 1
+
+// https://playwright.dev/docs/next/test-retries#reuse-single-page-between-tests
+let page: Page
+
+let chatList: Locator
+let selectedChats: Locator
+const getChat = (chatNum: number) =>
+  chatList.getByRole('tab', { name: `Some chat ${chatNum}` })
+const expectSelectedChats = async (chatNums: number[]) => {
+  await expect(selectedChats).toContainText(
+    chatNums.map(chatNum => `Some chat ${chatNum}`)
+  )
+}
+
+test.beforeAll(async ({ browser }) => {
+  const contextForProfileCreation = await browser.newContext()
+  const pageForProfileCreation = await contextForProfileCreation.newPage()
+  const pageForTestsP = browser.newPage()
+
+  await reloadPage(pageForProfileCreation)
+
+  existingProfiles =
+    (await loadExistingProfiles(pageForProfileCreation)) ?? existingProfiles
+
+  await createProfiles(
+    numberOfProfiles,
+    existingProfiles,
+    pageForProfileCreation,
+    contextForProfileCreation,
+    browser.browserType().name()
+  )
+
+  await contextForProfileCreation.close()
+  page = await pageForTestsP
+  await reloadPage(page)
+
+  // Let's stop at 9, so that we don't accidentally select "Chat 10"
+  // by providing the selector "Chat 1".
+  chatList = page.getByRole('tablist', { name: 'Chats' })
+  selectedChats = chatList.getByRole('tab', { selected: true })
+  await createNDummyChats(9)
+})
+
+test.afterAll(async ({ browser }) => {
+  await page.close()
+
+  const context = await browser.newContext()
+  const pageForProfileDeletion = await context.newPage()
+  await reloadPage(pageForProfileDeletion)
+  await deleteAllProfiles(pageForProfileDeletion, existingProfiles)
+  await context.close()
+})
+
+async function createNDummyChats(n: number) {
+  for (let i = 0; i < n; i++) {
+    await page.getByRole('button', { name: 'New Chat' }).click()
+    await page.getByRole('button', { name: 'New Group' }).click()
+    await page
+      .getByRole('textbox', { name: 'Group Name' })
+      .fill(`Some chat ${(i + 1).toString()}`)
+    await page.getByTestId('group-create-button').click()
+  }
+  await expect(chatList.getByRole('tab', { name: 'Some chat ' })).toContainText(
```

Comment:
Maybe comment here that the order of the strings in the array matters, since https://elaichenkov.github.io/playwright-expect/modules/tocontaintext.html says the parameter is of type string not string array?